custom_components/husqvarna_automower/camera.py

```diff
@@ -120,21 +211,79 @@ def supported_features(self) -> int:
         """Show supported features."""
         return CameraEntityFeature.ON_OFF
 
-    def _generate_image(self, data: dict):
-        position_history = AutomowerEntity.get_mower_attributes(self)["positions"]
-        location = (position_history[0]["latitude"], position_history[0]["longitude"])
+    def _find_image_scale(self):
+        """Find the scale ration in m/px and center of image."""
+        h_w = (
+            self._map_image.size[0],
+            self._map_image.size[1],
+        )  # Height/Width of image
+        self._c_img_px = int((0 + h_w[0]) / 2), int(
+            (0 + h_w[1]) / 2
+        )  # Center of image in pixels
+
+        # Center of image in lat/long
+        self._c_img_wgs84 = (
+            (self._top_left_coord[0] + self._bottom_right_coord[0]) / 2,
+            (self._top_left_coord[1] + self._bottom_right_coord[1]) / 2,
+        )
+
+        # Length of hypotenuse in meters
+        len_wgs84_m = geodesic(self._top_left_coord, self._bottom_right_coord).meters
+
+        # Length of hypotenuse in pixels
+        len_px = int(math.dist((0, 0), h_w))
+
+        self._px_meter = len_px / len_wgs84_m  # Scale in pixels/meter
+
+        _LOGGER.debug(
+            "Center px: %s, Center WGS84: %s, Len (m): %s, Len (px): %s, px/m: %s, Img HW (px): %s",
+            self._c_img_px,
+            self._c_img_wgs84,
+            len_wgs84_m,
+            len_px,
+            self._px_meter,
+            h_w,
+        )
+
+    def _calculate_update_frequency(self) -> None:
+        """Calculate the camera update frequecy."""
+        if self._last_update is not None:
+            update_frequency = (datetime.now() - self._last_update).seconds
+            if update_frequency > 0:
+                self._update_frequency = update_frequency
+                self._avg_update_frequency = (
+                    self._avg_update_frequency + self._update_frequency
+                ) / 2
+        self._last_update = datetime.now()
+
+    def _generate_image_cam(
+        self,
+        is_home: bool,
+        home_location: GpsPoint,
+        position_history: list,
+        mower_id: str,
+        path_color: list,
+        map_image: Image.Image,
+    ) -> None:
+        """Create image and mower overlay."""
+        if is_home and home_location:
+            location = home_location
+        else:
+            location = (
+                position_history[0]["latitude"],
+                position_history[0]["longitude"],
+            )
         if len(position_history) == 1:
-            self._position_history = position_history + self._position_history
-            position_history = self._position_history
+            self._position_history[mower_id] = (
+                position_history + self._position_history[mower_id]
```

Comment:
Is this a fresh install or was there an existing config entry?  I'll catch that error and add a test for it, but its odd to get an update but no existing position history for that mower.

Reply:
Existing condig, but I think it's really the quotation marks around `mower_id`?

Reply:
mower_id is a variable passed in to the function l, I have to track each position by mower Id now so we can have multiple mowers on a map, I'm wondering if since it's an existing config, that's not present, it's an easy fix, but a great call out also.  I'm testing on a new config, so never saw it.

Reply:
Latest commit fixes this, caused by getting a single coordinate position update with no previous position history.  We now account for it and there is a test added to replicate.